sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -1,205 +1,170 @@
+In v5:
 
-### Migrating code from `PartitionSender` to `EventHubProducerAsyncClient` for sending events to a partition
+```java
+public static void main(String[] args) throws Exception {
+    // This builds an async client. The sync version can be created by
+    // calling buildProducerClient() to create an EventHubProducerClient.
+    EventHubProducerAsyncClient client = new EventHubClientBuilder()
+        .connectionString("connection-string-for-an-event-hub")
+        .buildAsyncProducerClient();
 
-In v3, events could be published to a single partition using `PartitionSender`.
+    // Do things with the client.
+    // Dispose of client.
+    client.close();
+  }
+```
 
-In v5, this has been consolidated into a more efficient `send(EventDataBatch)` method. Batching merges information from
-multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at
-a time. Events are published to a specific partition [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions] is set
-before calling `createBatch(CreateBatchOptions)`.
+### Send events to an Event Hub
 
-The code below assumes all events fit into a single batch. For a more complete example, see sample: [Publishing events
-to specific partition][PublishEventsToSpecificPartition].
+In v3, events could be published to a single partition using `PartitionSender` or automatically routed to an available partition via `EventHubClient`.
 
-So in v3:
-```java
-EventHubClient client = EventHubClient.createFromConnectionStringSync(
-    "connection-string-for-an-event-hub", Executors.newScheduledThreadPool(4));
-List<EventData> events = Arrays.asList(EventData.create("foo".getBytes()));
+In v5, the **same client** can create an [EventDataBatchs][EventDataBatch] that routes to a specific partition or use automatic routing. Events are published to a specific partition using [`CreateBatchOptions.setPartitionId()`][CreateBatchOptions]; otherwise, the batch is automatically routed.
 
-CompletableFuture<Void> sendFuture = client.createPartitionSender("my-partition-id")
-    .thenCompose(producer -> {
-        EventDataBatch batch = producer.createBatch();
-        // Assuming all events fit into a single batch. This returns false if it does not.
-        // If it returns false, we send the batch, create another, and continue to add events.
-        for (EventData event : events) {
-            try {
-                batch.tryAdd(event);
-            } catch (PayloadSizeExceededException e) {
-                System.err.println("Event is larger than maximum allowed size. Exception: " + e);
-            }
-        }
+Additionally, we consolidate send into an efficient `send(EventDataBatch)` method on the producer client. Batching merges information from multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at a time.
```

Comment:
The batching happens even when list of events is passed instead of EventDataBatch. Assuming we go with my previous edit, we can change this as 

```suggestion
Batching merges information from multiple events into a single sent message, reducing the amount of network communication needed vs sending events one at a time.
```